src/main/java/com/atherys/quests/managers/LocationManager.java

```diff
@@ -37,17 +72,29 @@ protected LocationManager() {
 
         private String questId;
 
+        private boolean overlaps(QuestLocation questLocation){
```

Comment:
use Vector3d#distanceSquared() for better performance, sqrt is a heavier operation than squaring the radius you wish to compare against